server/src/main/java/com/objectcomputing/checkins/services/team/member/TeamMemberServicesImpl.java

```diff
@@ -108,4 +110,26 @@ public TeamMember update(@NotNull @Valid TeamMember teamMember) {
 
         return teamMembers;
     }
+
+    public void delete(@NotNull UUID id) {
+        MemberProfile currentUser = currentUserServices.getCurrentUser();
+        boolean isAdmin = currentUserServices.isAdmin();
+
+        TeamMember teamMember = teamMemberRepo.findById(id).orElse(null);
+        if(teamMember != null) {
+            Set<TeamMember> teamLeads = this.findByFields(teamMember.getTeamid(), null, true);
+
+            if (teamMemberRepo.findById(id).isEmpty()) {
+                throw new NotFoundException(String.format("Unable to locate teamMember to delete with id %s", id));
+            } else if (!isAdmin && teamLeads.stream().noneMatch(o -> o.getMemberid().equals(currentUser.getId()))) {
```

Comment:
Since checking if the authenticated user is a lead for a given user  comes up a lot this is something worth adding to the currentUserServices.

Reply:
I don't think this requested change is in the scope of this story, given that this logic is spread throughout the app currently and is fairly straight forward.  Further, I think I would be more prone to make it a method in the team or team member service, rather than current user and make it something that you could ask about any user id/team id combination...but that's certainly up for discussion/debate.

Regardless, I don't think we should hold this PR for that discussion and task. Thoughts, @meyerinj ?

Reply:
Probably, but let's do it in another story =)